ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -22,27 +22,31 @@
 import io.confluent.ksql.api.auth.AuthenticationPluginHandler;
 import io.confluent.ksql.api.auth.JaasAuthProvider;
 import io.confluent.ksql.api.auth.KsqlAuthorizationProviderHandler;
+import io.confluent.ksql.api.auth.RoleBasedAuthZHandler;
 import io.confluent.ksql.api.auth.SystemAuthenticationHandler;
 import io.confluent.ksql.rest.server.KsqlRestConfig;
 import io.confluent.ksql.security.KsqlSecurityExtension;
 import io.vertx.core.Handler;
 import io.vertx.core.http.ClientAuth;
-import io.vertx.ext.auth.AuthProvider;
 import io.vertx.ext.web.Router;
 import io.vertx.ext.web.RoutingContext;
-import io.vertx.ext.web.handler.AuthHandler;
 import io.vertx.ext.web.handler.BasicAuthHandler;
 import java.net.URI;
 import java.util.Optional;
 
 public final class AuthHandlers {
 
+  private static final String PROVIDER_KEY = "provider";
+  private enum Provider {
+    SYSTEM, JAAS, PLUGIN
```

Comment:
we select one of these methods of AuthN/Z - we use this enum to communicate across threads which ones we chose (it uses the context local data)

this allows us to login using JAAS separately from authorization, but still use the correct authZ mechanism